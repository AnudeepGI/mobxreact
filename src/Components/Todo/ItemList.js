import React from 'react';

const ItemList = (props) => {
    var templet = props.list.map((todo, key) => (
        <li key={key}>
            <input type="checkbox"
                onChange={props.onChange ? props.onChange.bind(this, todo) : undefined}
                value={todo.complete}
                checked={todo.complete}
            />
            {todo.value}
        </li>
    ));

    return templet;
}

export default ItemList;