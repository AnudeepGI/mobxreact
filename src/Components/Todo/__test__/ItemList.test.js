import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer';

import ItemList from '../ItemList'


it("rednder without crashing", () => {
    const { container } = render(<ItemList list={[{
        value : 'FirstVal1',
        id : 123,
        complete : false
    }]} onChange={(e) => {}} />)
    const li = container.querySelector('li');
    expect(li).toHaveTextContent("FirstVal1");
});

it("Match Snapshot - ItemList", () => {
   const tree = renderer.create(<ItemList list={[{
        value : 'FirstVal1',
        id : 123,
        complete : false
    }]} />).toJSON();
    expect(tree).toMatchSnapshot();
});