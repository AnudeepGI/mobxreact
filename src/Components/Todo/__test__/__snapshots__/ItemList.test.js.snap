// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match Snapshot - ItemList 1`] = `
<li>
  <input
    checked={false}
    type="checkbox"
    value={false}
  />
  FirstVal1
</li>
`;
